### DISCUSSION TITLE
Commits not appears on Github Activity Calendar

### DISCUSSION THREAD

**Discussion by CetinMutluOnal** [community/community#59080] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

In my projects, if i make a pull request from a branch to  another branch, it appears on my github actions but when i commit a change directly into a branch, it doesn't appears on my activity. I checked my local user configuration, e-mail addresses are ccorrect. Any idea for solve this problem ?

**Comment by leonden** [comment]
Do you have another account in use? I had this issue, with my Git user from work and then I had to add the email to my GitHub acocunt.

**Reply by CetinMutluOnal** [reply]
I have educational e-mail on GitHub for student promotions, but it's my secondary e-mail. My primary e-mail on GitHub and local configured e-mail on Git is same.

**Reply by leonden** [reply]
GitHub provides options to control the visibility of your activity. That also might be a problem. Ensure that your activity visibility settings are configured correctly. If I'm not mistaken you can find these settings by going to your GitHub profile, clicking on the "Settings" tab, selecting "Profile," and then checking the "Contributions" section.

**Reply by CetinMutluOnal** [reply]
Bro thanks for your helps but i already know what you said... My activity is working, but it dont show my direct commits, for example if a create new branch named develop from main branch and make some stuff inside of it, when i push and merge, i can see it on my activity. But if i make some stuff inside develop and just push it to develop branch on repo, i can't see them on activity. My problem occurs for directly commits, not pull requests or merging or etc.

**Reply by leonden** [reply]
Okay thanks for clarifying your problem. I'll try to investigate a little more :)

**Reply by leonden** [reply]
Perhaps you have branch protection settings on? It could also be possible, that the issue occurs due to webhook malfunction. I don't really have experience in this field but I've heard, that the GitHub actuvity feed relies on webhooks.

**Reply by CetinMutluOnal** [reply]
maybe you are right. but i haven't configured any branch to protected. I am also investigating but everybody who had same problem, they solved all with correct e-mail configuration. still desperate...

**Reply by leonden** [reply]
You could also setup a .gitconfig file in your local os user. I found a useful template [here](https://gist.github.com/pksunkara/988716).

**Reply by CetinMutluOnal** [reply]
It was good idea because i was never checked it before you, thanks for it. Now i check my git global config file, it had not any user fields like username, email etc. I configured them from template that you sent me and committed a test stuff, it doesn't work again...  Now my project git config and my global git config have my email but still not works...

**Reply by CetinMutluOnal** [reply, marked as answer]
Hi again, I found a some bot tool of github which checks a contribution. I gave one of my don't showing commits and it said me this :
![Ekran görüntüsü 2023-06-29 213614](https://github.com/community/community/assets/94856045/ccda7ada-b361-4af8-8b40-6e919bdb74fc)

Do you know anything about that?

**Reply by leonden** [reply]
Well based on the message it seems, that you committed in another branch than "main" or "gh-pages". Is it possible, that you have an unpublished local branch?

**Reply by CetinMutluOnal** [reply]
No, my branch is published on remote repo but it's not default branch. Message says main branch or gh-pages mandatory for commit activity but i think that's unreasonable 😀 In this situation we can't do anyting for it, it seems that's not problem, just a rule. Thanks for your help 🙏

**Reply by leonden** [reply]
Well good luck then ;)
